jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/UrlParameterDecoder.java

```diff
@@ -164,7 +165,7 @@ private boolean parseCompletely(CharIterator iter) throws IOException
         }
 
         complete();
-        return builder.hasCodingErrors();
+        return codingError;
```

Comment:
I think you need to set codingError to false at the start of this method, so the decoder can be reused.

Reply:
Done.